benchmarks/spiegelman_et_al_2016_gcubed/spiegelman_material.cc

```diff
@@ -0,0 +1,680 @@
+
+                              // compute the difference between the viscosity with and without the strain-rate difference.
+                              double viscosity_derivative = eta_component_plus - composition_viscosities[c];
+                              if (viscosity_derivative != 0)
+                                {
+                                  // when the difference is non-zero, divide by the difference.
+                                  viscosity_derivative /= std::max(edot_ii, min_strain_rate[c]*min_strain_rate[c]) * finite_difference_accuracy;
+                                }
+
+                              composition_viscosities_derivatives[c][strain_rate_indices] = viscosity_derivative;
+                            }
+
+                          /**
+                           * Now compute the finite difference derivative of the viscoisty to the pressure
+                           */
+                          double pressure_difference = in.pressure[i] + (std::fabs(in.pressure[i]) * finite_difference_accuracy);
+
+                          double pressure_difference_eta = compute_viscosity(edot_ii, pressure_difference,c,prefactor[c],true,min_visc[c],max_visc[c]);
+                          double deriv_pressure = pressure_difference_eta - composition_viscosities[c];
+
+
+                          if (pressure_difference_eta != 0)
+                            {
+                              if (in.pressure[i] != 0)
+                                {
+                                  deriv_pressure /= std::fabs(in.pressure[i]) * finite_difference_accuracy;
+                                }
+                              else
+                                {
+                                  deriv_pressure = 0;
+                                }
+                            }
+
+                          composition_dviscosities_dpressure[c] = deriv_pressure;
+                        }
+                    }
+                }
+              out.viscosities[i] = Utilities::weighted_p_norm_average(volume_fractions, composition_viscosities, viscosity_averaging_p);
+              Assert(dealii::numbers::is_finite(out.viscosities[i]),ExcMessage ("Error: Averaged viscosity is not finite."));
+
+              if (derivatives != NULL)
+                {
+                  derivatives->viscosity_derivative_wrt_strain_rate[i] = Utilities::derivative_of_weighted_p_norm_average(out.viscosities[i],volume_fractions, composition_viscosities, composition_viscosities_derivatives, viscosity_averaging_p);
+                  derivatives->viscosity_derivative_wrt_pressure[i] = Utilities::derivative_of_weighted_p_norm_average(out.viscosities[i],volume_fractions, composition_viscosities, composition_dviscosities_dpressure, viscosity_averaging_p);
+
+
+#ifdef DEBUG
+                  for (int x = 0; x < dim; x++)
+                    for (int y = 0; y < dim; y++)
```

Comment:
* use `unsigned int` here
* instead of `x,y`, we like to use `d,e` as variables to loop over dimensions
* say `++x` instead of `x++`.

same in line 443/444 below